release_docs/HISTORY-1_13.txt

```diff
@@ -538,7 +538,7 @@ New Features
       certain operations.
 
       These changes also introduce a new "ctl" callback
-      operation identified by the H5FD_CTL__MEM_COPY op code.
+      operation identified by the H5FD_CTL_MEM_COPY op code.
```

Comment:
They are [disallowed in C++](https://en.cppreference.com/w/c/language/identifier):

```
In file included from /Users/sean/external/hdf5/c++/src/H5CommonFG.cpp:16:
In file included from /Users/sean/external/hdf5/c++/src/H5Include.h:15:
In file included from /Users/sean/external/hdf5/src/hdf5.h:29:
/Users/sean/external/hdf5/src/H5FDpublic.h:191:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define H5FD_CTL__INVALID_OPCODE              0
        ^
```

Reply:
I see. This may be a bit trickier to handle since the macro names are already public, as you mentioned. Also, I believe that we stopped supporting compiling the C library with a C++ compiler a while ago due to some problems we were having in continuing to support both. That would be why this wasn't immediately seen at least.

Reply:
I kept the old spellings in the header file too.  Ideally the `#if 0` I have there for now would change to something allowing the one names to be defined or not. Other projects I work with do something like:

```
#if COMPILE_LEGACY_STUFF
#define OldSpelling NewSpelling
#endif
```

Does HDF5 have a similar mechanism?

Reply:
I tweaked my commit accordingly.

Other projects I work with have something like `H5_NO_DEPRECATED_SYMBOLS` but also something like `H5_FUTURE_NO_DEPRECATED_SYMBOLS`.  Then when a major release comes out, there's a mass rename of `H5_FUTURE_NO_DEPRECATED_SYMBOLS` to `H5_NO_DEPRECATED_SYMBOLS`.

Reply:
Regardless, this is a history document and the change should be reverted in this file.

Reply:
Reverted history change. Added small note in RELEASE.txt.